libbeat/publisher/queue/diskqueue/reader_loop.go

```diff
@@ -102,8 +102,9 @@ func (rl *readerLoop) processRequest(request readerLoopRequest) readerLoopRespon
 		return readerLoopResponse{err: err}
 	}
 	defer handle.Close()
-	_, err = handle.Seek(
-		segmentHeaderSize+int64(request.startOffset), os.SEEK_SET)
+	// getReader positions us at the start of the data region, so we use
+	// a relative seek to advance to the request position.
+	_, err = handle.Seek(int64(request.startOffset), os.SEEK_CUR)
```

Comment:
Well, it is common for a request to span a complete file, and for `startOffset` to be 0, though it is also nonzero when a segment is read and written at the same time (can only read up to what has already been written). But `startOffset` is absolute, not relative. This is a seek relative to a fresh file handle positioned at the start of the data region -- unlike the writer loop the reader loop does not cache file handles, and creates a fresh one for each request (basically because it is less to keep track of and the efficiency cost is minimal since we expect ~1 request per file at equilibrium).

Reply:
Thanks for clarifying.